### PR TITLE
chore: update RELEASES.md

### PR DESCRIPTION
**Pull request by kmill**
Reviewed all the commits in this release and updated the RELEASES.md file. Made an attempt to organize by rough themes to give a good overview of everything that's been done.

### CODE REVIEW

**Comment by semorrison** [comment]
Wow, @kmill, this is **really good**.

Only suggestion would be moving "Language features" to the top. Feels like the most impressive place to start.

**Comment by kmill** [comment]
Done!

File: RELEASES.md [hunk 1]
```diff
@@ -11,104 +11,406 @@ of each version.
...
+    bound, as in
+    ```
+    def Array.sum (arr : Array Nat) (i acc : Nat) : Nat :=
+      if _ : i < arr.size then
+        Array.sum arr (i+1) (acc + arr[i])
+      else
+        acc
+    ```
+    is recognized without having to say `termination_by arr.size - i`.
+    * [#3630](https://github.com/leanprover/lean4/pull/3630) makes `sizeOf` less necessary in termination arguments.
```
Comment (nomeata) [inline RELEASES.md:54]:
```suggestion
    * [#3630](https://github.com/leanprover/lean4/pull/3630) makes `termination_by?` not use `sizeOf` when not needed
```

File: RELEASES.md [hunk 2]
```diff
@@ -11,104 +11,406 @@ of each version.
...
+  * Docs: [#3704](https://github.com/leanprover/lean4/pull/3704).
+
+* DevOps
+  * [#3536](https://github.com/leanprover/lean4/pull/3536) and [#3833](https://github.com/leanprover/lean4/pull/3833)
+    add a checklist for the release process.
+  * [#3600](https://github.com/leanprover/lean4/pull/3600) runs nix-ci more uniformly.
+  * [#3612](https://github.com/leanprover/lean4/pull/3612) avoids argument limits when building on Windows.
+  * [#3682](https://github.com/leanprover/lean4/pull/3682) builds Lean's `.o` files in parallel to rest of core.
+  * [#3601](https://github.com/leanprover/lean4/pull/3601)
+    changes the way Lean is build on Windows (see breaking changes below).
```
Comment (kmill) [inline RELEASES.md:387]:
```suggestion
    changes the way Lean is built on Windows (see breaking changes below).
```

**Pull request merged**